packages/sysweb3-keyring/src/transactions/ethereum.ts

```diff
@@ -244,6 +263,56 @@ export class EthereumTransactions implements IEthereumTransactions {
   //TODO: This function needs to be refactored
   sendFormattedTransaction = async (params: SimpleTransactionRequest) => {
     const { decryptedPrivateKey } = this.getDecryptedPrivateKey();
+    const { activeAccountType, activeAccountId, accounts, activeNetwork } =
+      this.getState();
+    const activeAccount = accounts[activeAccountType][activeAccountId];
+    const transactionNonce = await this.getRecommendedNonce(
+      activeAccount.address
+    );
+    //TODO: split into two separate sub functions one for trezor and other for HD and import accounts
+    const formatParams = omit(params, 'from'); //From is not needed we're already passing in the HD derivation path so it can be inferred
+    const txFormattedForTrezor = {
+      ...formatParams,
+      // @ts-ignore
+      gasLimit: `${params.gasLimit.toHexString()}`,
+      // @ts-ignore
+      maxFeePerGas: `${params.maxFeePerGas.toHexString()}`,
+      // @ts-ignore
+      maxPriorityFeePerGas: `${params.maxPriorityFeePerGas.toHexString()}`,
+      // @ts-ignore
+      value: `${params.value.toHexString()}`,
+      nonce: this.toBigNumber(transactionNonce)._hex,
+      chainId: activeNetwork.chainId,
+    };
+
+    if (activeAccountType === KeyringAccountType.Trezor) {
```

Comment:
This verification should be done before, we need to  split sendFormattedTransaction into two subfunctions, one for Imported and HDAccount. Other for Trezor.
That way we avoid doing not needed computation for trezor before the time. We can get the parameters that will be needed for both before routing to one or other function and passing them as arguments something like this:
```js
sendFormattedTransaction = async (params: SimpleTransactionRequest) => {
    const { decryptedPrivateKey } = this.getDecryptedPrivateKey();
    ... fetch needed parameters by both
    switch (activeAccountType):
    case 0: 
    function0( neededParams);
    break;
    case 1: 
    function1( neededParams);
    break;
    case ...
    default:
    
    ...
    
 
```